libminifi/src/core/extension/ExtensionManager.cpp

```diff
@@ -0,0 +1,136 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "core/extension/ExtensionManager.h"
+#include "core/logging/LoggerConfiguration.h"
+#include "utils/file/FileUtils.h"
+#include "core/extension/Executable.h"
+#include "utils/file/FilePattern.h"
+#include "core/extension/DynamicLibrary.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace core {
+namespace extension {
+
+namespace {
+
+struct LibraryDescriptor {
+  std::string name;
+  std::filesystem::path dir;
+  std::string filename;
+
+  bool verify(const std::shared_ptr<logging::Logger>& /*logger*/) const {
+    // TODO(adebreceni): check signature
+    return true;
+  }
+
+  std::filesystem::path getFullPath() const {
+    return dir / filename;
+  }
```

Comment:
Consider marking getters `[[nodiscard]]`. Clang-tidy warns me for all of these with modernize-use-nodiscard through CLion.

Reply:
added